### PR TITLE
More precisly define which dhcp message types we are listening for

### PR DESCRIPTION
**Pull request by jrouzierinverse**

### CODE REVIEW

File: NEWS.asciidoc [hunk 1]
```diff
@@ -21,6 +21,7 @@ New Features
 Enhancements
 ++++++++++++
 * Added httpd.aaa service as a new API service for the exclusive use of RADIUS
+* More precisly define which dhcp message types we are listening for
```
Comment (extrafu) [inline NEWS.asciidoc]: typo -> precisely

File: html/pfappserver/lib/pfappserver/Form/Config/Pf.pm [hunk 2]
```diff
@@ -43,6 +43,8 @@ sub field_list {
             label => $doc_section_name,
           };
         my $type = $doc_section->{type} || "text";
+        #skip if hidden
```
Comment (extrafu) [inline html/pfappserver/lib/pfappserver/Form/Config/Pf.pm:46]: Remove if unused.

Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Form/Config/Pf.pm:46]: It is used

File: lib/pf/util/dhcp.pm [hunk 3]
```diff
@@ -289,6 +289,25 @@ sub _decode_dhcp_option82 {
...
+=cut
+
+sub make_pcap_filter {
+    my (@types) = @_;
+    #listen to all if no types are provided
+    return "udp and (port 67 or port 68)" unless @types;
+    for my $type (@types) {
+       die "Unknown message type $type" unless exists $MESSAGE_TYPE{$type} && defined $MESSAGE_TYPE{$type};
+    }
+    my $type_filter = join(" or ",map { sprintf("(udp[250:1] = 0x%x)",$MESSAGE_TYPE{$_}) } @types);
```
Comment (fdurand) [inline lib/pf/util/dhcp.pm:306]: If i have only one type, does the filter will look like : ( (port 67 or port 68) and ( or udp[250:1] = 0x5) ) ?

Comment (jrouzierinverse) [inline lib/pf/util/dhcp.pm:306]: Yes
perl -Ilib -Mpf::util::dhcp -e 'print make_pcap_filter(qw(DHCPDISCOVER)),"\n"'
(port 67 or port 68) and ( (udp[250:1] = 0x5) )

Comment (fdurand) [inline lib/pf/util/dhcp.pm:306]: Ok so ready to merge

File: lib/pf/config.pm [hunk 4]
```diff
@@ -653,6 +653,7 @@ sub readPfConfigFiles {
                         )
                     ];
                 }
+                $Config{network}{dhcp_filter_by_message_types} = [split(/\s*,\s*/,$Config{network}{dhcp_filter_by_message_types} || '')],
```
Comment (jrouzierinverse) [inline lib/pf/config.pm:656]: If we want to further fine tune the filtering.
We could remove the following conditionally 
- The offer message if rogue detection is disable.
- The request message if rogue detection is disable and there is no firewallsso configured and not inline mode

File: lib/pf/util/dhcp.pm (hunk 3 above)
Comment (jrouzierinverse) [inline lib/pf/util/dhcp.pm:306]: Then push button

File: lib/pf/config.pm (hunk 4 above)
Comment (fdurand) [inline lib/pf/config.pm:656]: For setup with high dhcp traffic it can be very usefull

**Pull request merged**